storage: TestStoreRangeRemoveDead failed under stress [skipped]

SHA: https://github.com/cockroachdb/cockroach/commits/1c552ab65b29535c9f2b48516ac8ba03cb7235f4

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1098703&tab=buildLog

```
I190117 09:09:48.613035 61723 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1 (n3,s3):6 (n4,s4):3 (n2,s2):4]
I190117 09:09:49.616277 61723 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1 (n3,s3):6 (n4,s4):3 (n2,s2):4]
I190117 09:09:50.636347 61723 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1 (n3,s3):6 (n4,s4):3 (n2,s2):4]
I190117 09:09:51.638780 61723 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1 (n3,s3):6 (n4,s4):3 (n2,s2):4]
I190117 09:09:52.643471 61723 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1 (n3,s3):6 (n4,s4):3 (n2,s2):4]
I190117 09:09:53.654118 61723 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1 (n3,s3):6 (n4,s4):3 (n2,s2):4]
I190117 09:09:54.687878 61723 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1 (n3,s3):6 (n4,s4):3 (n2,s2):4]
I190117 09:09:55.706096 61723 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1 (n3,s3):6 (n4,s4):3 (n2,s2):4]
I190117 09:09:56.752434 61723 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1 (n3,s3):6 (n4,s4):3 (n2,s2):4]
I190117 09:09:57.755770 61723 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1 (n3,s3):6 (n4,s4):3 (n2,s2):4]
I190117 09:09:58.759706 61723 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1 (n3,s3):6 (n4,s4):3 (n2,s2):4]
I190117 09:09:59.763976 61723 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1 (n3,s3):6 (n4,s4):3 (n2,s2):4]
I190117 09:10:00.768668 61723 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1 (n3,s3):6 (n4,s4):3 (n2,s2):4]
I190117 09:10:01.786086 61723 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1 (n3,s3):6 (n4,s4):3 (n2,s2):4]
I190117 09:10:02.798985 61723 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1 (n3,s3):6 (n4,s4):3 (n2,s2):4]
I190117 09:10:03.833962 61723 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1 (n3,s3):6 (n4,s4):3 (n2,s2):4]
I190117 09:10:04.847974 61723 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1 (n3,s3):6 (n4,s4):3 (n2,s2):4]
I190117 09:10:05.853642 61723 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1 (n3,s3):6 (n4,s4):3 (n2,s2):4]
I190117 09:10:06.864044 61723 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1 (n3,s3):6 (n4,s4):3 (n2,s2):4]
W190117 09:10:07.917408 62050 storage/store.go:1465  [s3,r1/6:/M{in-ax}] could not gossip first range descriptor: node unavailable; try another peer
W190117 09:10:07.991450 62108 gossip/gossip.go:1520  [n5] no incoming or outgoing connections
W190117 09:10:07.992503 62460 storage/raft_transport.go:583  while processing outgoing Raft queue to node 4: EOF:
--- FAIL: TestStoreRangeRemoveDead (45.75s)
    <autogenerated>:1: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1 (n3,s3):6 (n4,s4):3 (n2,s2):4]
        goroutine 61723 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc000b37f20, 0x4925c60)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4997400, 0xc001860700, 0xc000b37ef0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc001860700)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:3101 +0xaef
        testing.tRunner(0xc001860700, 0x3c88908)
        	/usr/local/go/src/testing/testing.go:827 +0x163
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/b52b7d1382b454ce1bb43f2187088aef9c557ed5

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1102014&tab=buildLog

```
I190120 09:30:01.722466 61625 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190120 09:30:02.725257 61625 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190120 09:30:03.733065 61625 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190120 09:30:04.736977 61625 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190120 09:30:05.740880 61625 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190120 09:30:06.744876 61625 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190120 09:30:07.749985 61625 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190120 09:30:08.770999 61625 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190120 09:30:09.803979 61625 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190120 09:30:10.807359 61625 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190120 09:30:11.826541 61625 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190120 09:30:12.842713 61625 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190120 09:30:13.859320 61625 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190120 09:30:14.862529 61625 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190120 09:30:15.905398 61625 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190120 09:30:16.917463 61625 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190120 09:30:17.944326 61625 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190120 09:30:18.949843 61625 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190120 09:30:19.970405 61625 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190120 09:30:20.976683 61625 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190120 09:30:21.991200 61625 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190120 09:30:23.002017 61625 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190120 09:30:24.008840 61625 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190120 09:30:25.036788 61625 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190120 09:30:26.042466 61625 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190120 09:30:27.047176 61625 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190120 09:30:28.051575 61625 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
W190120 09:30:29.160570 62373 gossip/gossip.go:1520  [n6] no incoming or outgoing connections
--- FAIL: TestStoreRangeRemoveDead (46.19s)
    <autogenerated>:1: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1]
        goroutine 61625 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc000c0c450, 0x4918e80)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x498a540, 0xc0010ece00, 0xc000c0c420)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc0010ece00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:3101 +0xaef
        testing.tRunner(0xc0010ece00, 0x3c814d8)
        	/usr/local/go/src/testing/testing.go:827 +0x163
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/798304879367166c8954825f40c404ba100cea0a

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1103973&tab=buildLog

```
I190122 07:06:06.354324 61614 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1 (n2,s2):2 (n5,s5):3 (n4,s4):4]
I190122 07:06:07.360875 61614 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1 (n2,s2):2 (n5,s5):3 (n4,s4):4]
I190122 07:06:08.402444 61614 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1 (n2,s2):2 (n5,s5):3 (n4,s4):4]
I190122 07:06:09.431798 61614 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1 (n2,s2):2 (n5,s5):3 (n4,s4):4]
I190122 07:06:10.443370 61614 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1 (n2,s2):2 (n5,s5):3 (n4,s4):4]
I190122 07:06:11.469239 61614 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1 (n2,s2):2 (n5,s5):3 (n4,s4):4]
I190122 07:06:12.491608 61614 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1 (n2,s2):2 (n5,s5):3 (n4,s4):4]
I190122 07:06:13.547085 61614 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1 (n2,s2):2 (n5,s5):3 (n4,s4):4]
I190122 07:06:14.596743 61614 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1 (n2,s2):2 (n5,s5):3 (n4,s4):4]
I190122 07:06:15.599982 61614 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1 (n2,s2):2 (n5,s5):3 (n4,s4):4]
I190122 07:06:16.602779 61614 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1 (n2,s2):2 (n5,s5):3 (n4,s4):4]
I190122 07:06:17.607253 61614 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1 (n2,s2):2 (n5,s5):3 (n4,s4):4]
I190122 07:06:18.621545 61614 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1 (n2,s2):2 (n5,s5):3 (n4,s4):4]
I190122 07:06:19.626384 61614 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1 (n2,s2):2 (n5,s5):3 (n4,s4):4]
I190122 07:06:20.634160 61614 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1 (n2,s2):2 (n5,s5):3 (n4,s4):4]
I190122 07:06:21.669202 61614 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1 (n2,s2):2 (n5,s5):3 (n4,s4):4]
I190122 07:06:22.691358 61614 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1 (n2,s2):2 (n5,s5):3 (n4,s4):4]
I190122 07:06:23.716333 61614 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1 (n2,s2):2 (n5,s5):3 (n4,s4):4]
I190122 07:06:24.719000 61614 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1 (n2,s2):2 (n5,s5):3 (n4,s4):4]
I190122 07:06:25.781293 61614 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1 (n2,s2):2 (n5,s5):3 (n4,s4):4]
W190122 07:06:26.811229 62233 storage/store.go:1465  [s5,r1/3:/M{in-ax}] could not gossip first range descriptor: node unavailable; try another peer
W190122 07:06:26.818571 61880 storage/store.go:1465  [s2,r1/2:/M{in-ax}] could not gossip first range descriptor: node unavailable; try another peer
--- FAIL: TestStoreRangeRemoveDead (45.89s)
    <autogenerated>:1: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1 (n2,s2):2 (n5,s5):3 (n4,s4):4]
        goroutine 61614 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc000f20e70, 0x4905200)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x49765c0, 0xc000fe1b00, 0xc000f20b40)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc000fe1b00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:3101 +0xaef
        testing.tRunner(0xc000fe1b00, 0x3c7c820)
        	/usr/local/go/src/testing/testing.go:827 +0x163
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/f37d09cd3cdd32f4d4894611cfd60caf25c10fff

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1105906&tab=buildLog

```
I190123 08:33:36.650879 61542 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190123 08:33:37.654423 61542 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190123 08:33:38.665477 61542 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190123 08:33:39.669301 61542 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190123 08:33:40.679975 61542 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190123 08:33:41.693631 61542 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190123 08:33:42.713720 61542 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190123 08:33:43.753617 61542 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190123 08:33:44.756142 61542 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190123 08:33:45.772727 61542 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190123 08:33:46.776927 61542 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190123 08:33:47.779497 61542 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190123 08:33:48.798853 61542 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190123 08:33:49.820354 61542 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190123 08:33:50.823553 61542 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190123 08:33:51.827334 61542 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190123 08:33:52.860163 61542 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190123 08:33:53.870647 61542 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190123 08:33:54.888940 61542 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190123 08:33:55.895706 61542 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190123 08:33:56.908214 61542 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190123 08:33:57.916997 61542 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190123 08:33:58.922310 61542 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190123 08:33:59.959687 61542 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190123 08:34:00.962901 61542 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190123 08:34:01.965200 61542 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190123 08:34:02.986796 61542 storage/client_raft_test.go:3101  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
W190123 08:34:04.069054 61870 gossip/gossip.go:1520  [n4] no incoming or outgoing connections
--- FAIL: TestStoreRangeRemoveDead (46.31s)
    <autogenerated>:1: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1]
        goroutine 61542 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc00149d110, 0x4916680)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4987c60, 0xc0013ad200, 0xc00149d020)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc0013ad200)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:3101 +0xaef
        testing.tRunner(0xc0013ad200, 0x3c8a050)
        	/usr/local/go/src/testing/testing.go:827 +0x163
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/e09c33ffc49f15b4909d4f4509b96d1c3c36a956

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1125435&tab=buildLog

```
I190205 10:02:29.230897 61676 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190205 10:02:30.247291 61676 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190205 10:02:31.250657 61676 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190205 10:02:32.262961 61676 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190205 10:02:33.266247 61676 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190205 10:02:34.295928 61676 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190205 10:02:35.303696 61676 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190205 10:02:36.308157 61676 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190205 10:02:37.312537 61676 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190205 10:02:38.323934 61676 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190205 10:02:39.331636 61676 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190205 10:02:40.334038 61676 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190205 10:02:41.340681 61676 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190205 10:02:42.344446 61676 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190205 10:02:43.363331 61676 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190205 10:02:44.366821 61676 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190205 10:02:45.380138 61676 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190205 10:02:46.386742 61676 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190205 10:02:47.414245 61676 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190205 10:02:48.421213 61676 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190205 10:02:49.437122 61676 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
W190205 10:02:50.496621 62292 internal/client/txn.go:520  [hb] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (server shutdown)
I190205 10:02:50.499737 62292 storage/node_liveness.go:773  [hb] retrying liveness update after storage.errRetryLiveness: result is ambiguous (server shutdown)
W190205 10:02:50.502207 62292 internal/client/txn.go:520  [hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W190205 10:02:50.503211 62292 storage/node_liveness.go:452  [hb] failed node liveness heartbeat: node unavailable; try another peer
W190205 10:02:50.605532 61720 gossip/gossip.go:1496  [n3] no incoming or outgoing connections
--- FAIL: TestStoreRangeRemoveDead (46.49s)
    <autogenerated>:1: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1]
        goroutine 61676 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc00168f560, 0x49592a0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x49cb0c0, 0xc000c46000, 0xc00168f530)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc000c46000)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:3091 +0xadb
        testing.tRunner(0xc000c46000, 0x3cb9ba0)
        	/usr/local/go/src/testing/testing.go:827 +0x163
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/4fc4d63ddddeb507564bfe53c6cd79549ff2fd27

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1142122&tab=buildLog

```
I190216 09:57:22.354282 62911 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190216 09:57:23.381571 62911 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190216 09:57:24.383761 62911 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190216 09:57:25.411610 62911 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190216 09:57:26.436177 62911 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190216 09:57:27.438455 62911 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190216 09:57:28.482369 62911 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190216 09:57:29.485118 62911 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190216 09:57:30.489715 62911 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190216 09:57:31.493053 62911 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190216 09:57:32.495592 62911 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190216 09:57:33.535068 62911 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190216 09:57:34.542401 62911 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190216 09:57:35.544586 62911 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190216 09:57:36.552036 62911 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190216 09:57:37.562156 62911 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190216 09:57:38.600568 62911 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190216 09:57:39.607849 62911 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190216 09:57:40.623441 62911 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190216 09:57:41.645437 62911 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190216 09:57:42.648879 62911 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190216 09:57:43.651333 62911 storage/client_raft_test.go:3091  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190216 09:57:44.655610 63750 util/stop/stopper.go:547  quiescing; tasks left:
1      [async] intent_resolver_batcher
I190216 09:57:44.656661 63697 util/stop/stopper.go:547  quiescing; tasks left:
1      [async] intent_resolver_batcher
I190216 09:57:44.656999 63747 util/stop/stopper.go:547  quiescing; tasks left:
1      [async] intent_resolver_batcher
I190216 09:57:44.659368 63698 util/stop/stopper.go:547  quiescing; tasks left:
1      [async] intent_resolver_batcher
I190216 09:57:44.659621 63749 util/stop/stopper.go:547  quiescing; tasks left:
1      [async] intent_resolver_batcher
I190216 09:57:44.659838 63748 util/stop/stopper.go:547  quiescing; tasks left:
1      [async] intent_resolver_batcher
--- FAIL: TestStoreRangeRemoveDead (45.74s)
    <autogenerated>:1: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1]
        goroutine 62911 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc0019a2d20, 0x49e6ae0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4a59b20, 0xc000be6f00, 0xc0019a2cf0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc000be6f00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:3091 +0xaef
        testing.tRunner(0xc000be6f00, 0x3d2f8a8)
        	/usr/local/go/src/testing/testing.go:827 +0x163
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/032c4980720abc1bdd71e4428e4111e6e6383297

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1159369&tab=buildLog

```
I190302 08:05:59.695400 63032 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190302 08:06:00.705503 63032 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190302 08:06:01.742940 63032 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190302 08:06:02.755469 63032 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190302 08:06:03.759797 63032 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190302 08:06:04.786375 63032 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190302 08:06:05.789934 63032 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190302 08:06:06.801728 63032 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190302 08:06:07.820909 63032 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190302 08:06:08.829993 63032 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190302 08:06:09.842441 63032 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190302 08:06:10.857423 63032 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190302 08:06:11.888398 63032 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190302 08:06:12.893284 63032 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190302 08:06:13.901955 63032 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190302 08:06:14.911648 63032 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190302 08:06:15.913563 63854 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190302 08:06:15.913925 63854 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
I190302 08:06:15.914246 63851 util/stop/stopper.go:546  quiescing; tasks left:
1      mtc send
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190302 08:06:15.914406 63852 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190302 08:06:15.914935 63855 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190302 08:06:15.915380 63853 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190302 08:06:15.915610 63851 util/stop/stopper.go:546  quiescing; tasks left:
1      mtc send
I190302 08:06:15.915638 63852 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190302 08:06:15.916677 63856 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
--- FAIL: TestStoreRangeRemoveDead (46.04s)
    <autogenerated>:1: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1]
        goroutine 63032 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc0008af260, 0x4a3b880)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4aae9e0, 0xc0015fae00, 0xc0008af200)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc0015fae00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2944 +0xadb
        testing.tRunner(0xc0015fae00, 0x3d6f848)
        	/usr/local/go/src/testing/testing.go:827 +0x163
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/38bb1e7905b89f911bd74be4f5830217ffb7b343

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1169225&tab=buildLog

```
I190309 07:27:14.093970 214603 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190309 07:27:15.147901 214603 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190309 07:27:16.152727 214603 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190309 07:27:17.254986 214603 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190309 07:27:18.282513 214603 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190309 07:27:19.293498 214603 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190309 07:27:20.340263 214603 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190309 07:27:21.366884 214603 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190309 07:27:22.408473 214603 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190309 07:27:23.431029 224492 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190309 07:27:23.434425 224492 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190309 07:27:23.437037 224494 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190309 07:27:23.439204 224495 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190309 07:27:23.449869 224490 util/stop/stopper.go:546  quiescing; tasks left:
2      mtc send
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190309 07:27:23.450715 224491 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190309 07:27:23.451033 224491 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190309 07:27:23.451303 224493 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190309 07:27:23.452365 224490 util/stop/stopper.go:546  quiescing; tasks left:
2      mtc send
I190309 07:27:23.472005 224490 util/stop/stopper.go:546  quiescing; tasks left:
1      mtc send
W190309 07:27:23.495717 214758 internal/client/txn.go:509  [hb] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (server shutdown)
I190309 07:27:23.496823 214758 storage/node_liveness.go:775  [hb] retrying liveness update after storage.errRetryLiveness: result is ambiguous (server shutdown)
W190309 07:27:23.500413 214758 internal/client/txn.go:509  [hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W190309 07:27:23.500788 214758 storage/node_liveness.go:458  [hb] failed node liveness heartbeat: node unavailable; try another peer
W190309 07:27:23.866845 215188 gossip/gossip.go:1501  [n4] first range unavailable; trying remaining resolvers
--- FAIL: TestStoreRangeRemoveDead (46.96s)
    <autogenerated>:1: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1]
        goroutine 214603 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc00aa8a3f0, 0x4a6efa0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4ae2820, 0xc004a66300, 0xc00aa8a3c0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc004a66300)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2944 +0xadb
        testing.tRunner(0xc004a66300, 0x3d86670)
        	/usr/local/go/src/testing/testing.go:827 +0x163
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/5f7de3c35348e847dc0d4ab1ba8d76574e1706c2

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1170452&tab=buildLog

```
I190310 07:46:50.437765 216195 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190310 07:46:51.453210 216195 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190310 07:46:52.469916 216195 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190310 07:46:53.483300 216195 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190310 07:46:54.487198 216195 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190310 07:46:55.492991 216195 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190310 07:46:56.510864 216195 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190310 07:46:57.516151 216195 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190310 07:46:58.540867 216195 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190310 07:46:59.586033 216195 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190310 07:47:00.599446 216195 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190310 07:47:01.649045 216195 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190310 07:47:02.668521 226388 util/stop/stopper.go:546  quiescing; tasks left:
1      mtc send
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190310 07:47:02.668984 226388 util/stop/stopper.go:546  quiescing; tasks left:
1      mtc send
1      [async] intent_resolver_gc_batcher
I190310 07:47:02.669386 226388 util/stop/stopper.go:546  quiescing; tasks left:
1      mtc send
I190310 07:47:02.670037 226393 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190310 07:47:02.670593 226389 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190310 07:47:02.670851 226389 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190310 07:47:02.672013 226391 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190310 07:47:02.674322 226392 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190310 07:47:02.697361 226392 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190310 07:47:02.697612 226390 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
W190310 07:47:02.738930 216366 storage/store.go:1485  [s1,r1/1:/M{in-ax}] could not gossip first range descriptor: node unavailable; try another peer
W190310 07:47:02.996841 217274 gossip/gossip.go:1496  [n6] no incoming or outgoing connections
W190310 07:47:02.998389 216853 gossip/gossip.go:1496  [n4] no incoming or outgoing connections
--- FAIL: TestStoreRangeRemoveDead (47.19s)
    <autogenerated>:1: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1]
        goroutine 216195 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc000c9f7d0, 0x4a73e40)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4ae7780, 0xc0010c2100, 0xc000c9f7a0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc0010c2100)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2944 +0xadb
        testing.tRunner(0xc0010c2100, 0x3d8a710)
        	/usr/local/go/src/testing/testing.go:827 +0x163
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/a119a3a158725c9e3f9b8084d9398601c0e67007

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1171269&tab=buildLog

```
I190311 07:24:26.881730 213280 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190311 07:24:27.885338 213280 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190311 07:24:28.893212 213280 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190311 07:24:29.916225 213280 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190311 07:24:30.919748 213280 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190311 07:24:31.958695 213280 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190311 07:24:33.006187 213280 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190311 07:24:34.055320 213280 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190311 07:24:35.079903 213280 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190311 07:24:36.107817 213280 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190311 07:24:37.150431 213280 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190311 07:24:38.155414 213280 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190311 07:24:39.195477 213280 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190311 07:24:40.205268 213280 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190311 07:24:41.247008 213280 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190311 07:24:42.270596 213280 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190311 07:24:43.280391 223616 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190311 07:24:43.281687 223611 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190311 07:24:43.282411 223612 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190311 07:24:43.282668 223612 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
I190311 07:24:43.283376 223614 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190311 07:24:43.283721 223614 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190311 07:24:43.284302 223613 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190311 07:24:43.284597 223613 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
I190311 07:24:43.295759 223615 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
--- FAIL: TestStoreRangeRemoveDead (47.13s)
    <autogenerated>:1: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1]
        goroutine 213280 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc0002d80f0, 0x4a744a0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4ae7de0, 0xc00198a000, 0xc0002d80c0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc00198a000)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2944 +0xadb
        testing.tRunner(0xc00198a000, 0x3d8b098)
        	/usr/local/go/src/testing/testing.go:827 +0x163
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/9d058d53c8a82fceb2205f1827c26f1bf36c32ba

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1173350&tab=buildLog

```
I190312 08:26:05.750971 216824 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190312 08:26:06.829380 216824 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190312 08:26:07.919024 216824 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190312 08:26:08.949130 216824 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190312 08:26:09.957979 216824 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190312 08:26:10.979077 216824 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190312 08:26:12.015898 216824 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190312 08:26:13.068143 216824 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190312 08:26:14.119027 216824 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190312 08:26:15.129278 226717 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190312 08:26:15.131881 226715 util/stop/stopper.go:546  quiescing; tasks left:
1      mtc send
1      [async] storage.pendingLeaseRequest: requesting lease
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190312 08:26:15.137276 226716 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190312 08:26:15.137615 226716 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190312 08:26:15.137865 226718 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190312 08:26:15.138126 226719 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190312 08:26:15.138376 226719 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
I190312 08:26:15.138658 226720 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
W190312 08:26:15.141730 217232 internal/client/txn.go:509  [hb] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (server shutdown)
I190312 08:26:15.146486 217232 storage/node_liveness.go:775  [hb] retrying liveness update after storage.errRetryLiveness: result is ambiguous (server shutdown)
W190312 08:26:15.148937 217232 internal/client/txn.go:509  [hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W190312 08:26:15.160785 217232 storage/node_liveness.go:458  [hb] failed node liveness heartbeat: node unavailable; try another peer
I190312 08:26:15.171786 226718 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
W190312 08:26:15.343622 217231 gossip/gossip.go:1496  [n3] no incoming or outgoing connections
--- FAIL: TestStoreRangeRemoveDead (46.55s)
    <autogenerated>:1: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1]
        goroutine 216824 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc001157290, 0x4a7a100)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4aeda40, 0xc0018c4800, 0xc001157260)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc0018c4800)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2944 +0xadb
        testing.tRunner(0xc0018c4800, 0x3d8ebe8)
        	/usr/local/go/src/testing/testing.go:827 +0x163
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/e9ec5a3d827b6b4620b8aeea1851193453ca952c

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1175767&tab=buildLog

```
I190313 09:05:51.468978 213544 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190313 09:05:52.480371 213544 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190313 09:05:53.502567 213544 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190313 09:05:54.536166 213544 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190313 09:05:55.575169 213544 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190313 09:05:56.578852 213544 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190313 09:05:57.603413 213544 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190313 09:05:58.636332 213544 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190313 09:05:59.653678 213544 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190313 09:06:00.663088 223697 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190313 09:06:00.663461 223698 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190313 09:06:00.663572 223696 util/stop/stopper.go:546  quiescing; tasks left:
2      mtc send
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190313 09:06:00.663905 223699 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190313 09:06:00.664348 223700 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190313 09:06:00.664655 223700 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190313 09:06:00.664940 223749 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190313 09:06:00.663937 223696 util/stop/stopper.go:546  quiescing; tasks left:
2      mtc send
1      [async] intent_resolver_gc_batcher
I190313 09:06:00.666835 223749 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
W190313 09:06:00.679420 213738 internal/client/txn.go:509  [hb] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (server shutdown)
I190313 09:06:00.680379 213738 storage/node_liveness.go:775  [hb] retrying liveness update after storage.errRetryLiveness: result is ambiguous (server shutdown)
I190313 09:06:00.683907 223696 util/stop/stopper.go:546  quiescing; tasks left:
1      mtc send
1      [async] intent_resolver_gc_batcher
W190313 09:06:00.695904 213738 internal/client/txn.go:509  [hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W190313 09:06:00.696277 213738 storage/node_liveness.go:458  [hb] failed node liveness heartbeat: node unavailable; try another peer
--- FAIL: TestStoreRangeRemoveDead (46.87s)
    <autogenerated>:1: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1]
        goroutine 213544 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc0004b3fb0, 0x4a5f040)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4ad2980, 0xc001918300, 0xc0004b3e60)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc001918300)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2944 +0xadb
        testing.tRunner(0xc001918300, 0x3d98508)
        	/usr/local/go/src/testing/testing.go:827 +0x163
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/57e825a7940495b67e0cc7213a5fabc24e12be0e

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1177426&tab=buildLog

```
I190314 07:04:21.834899 217538 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190314 07:04:22.908478 217538 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190314 07:04:23.939197 217538 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190314 07:04:24.968687 217538 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190314 07:04:25.987389 217538 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190314 07:04:27.013711 217538 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190314 07:04:28.025995 217538 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190314 07:04:29.068870 217538 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190314 07:04:30.074988 217538 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190314 07:04:31.079473 217538 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190314 07:04:32.096034 217538 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190314 07:04:33.099870 217538 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190314 07:04:34.170125 217538 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190314 07:04:35.236334 217538 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190314 07:04:36.238396 226842 util/stop/stopper.go:546  quiescing; tasks left:
1      mtc send
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190314 07:04:36.239117 226845 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190314 07:04:36.239624 226846 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190314 07:04:36.240335 226846 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190314 07:04:36.240674 226847 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190314 07:04:36.241715 226845 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
I190314 07:04:36.238773 226844 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190314 07:04:36.238883 226843 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190314 07:04:36.252728 226844 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
W190314 07:04:36.494180 218157 gossip/gossip.go:1496  [n4] no incoming or outgoing connections
W190314 07:04:36.496613 217832 gossip/gossip.go:1496  [n2] no incoming or outgoing connections
--- FAIL: TestStoreRangeRemoveDead (46.62s)
    <autogenerated>:1: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1]
        goroutine 217538 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc000e15590, 0x4a630e0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4ad6660, 0xc001568200, 0xc000e15560)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc001568200)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2944 +0xadb
        testing.tRunner(0xc001568200, 0x3d9bff8)
        	/usr/local/go/src/testing/testing.go:827 +0x163
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/70e3468e7ed5fb30b10eaaf972acbb0f16099d01

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1179874&tab=buildLog

```
I190315 08:41:34.148167 207233 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190315 08:41:35.179759 207233 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190315 08:41:36.201651 207233 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190315 08:41:37.214945 207233 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190315 08:41:38.246703 207233 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190315 08:41:39.268865 207233 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190315 08:41:40.296865 207233 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190315 08:41:41.303412 207233 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190315 08:41:42.329992 207233 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190315 08:41:43.341389 207233 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190315 08:41:44.450126 207233 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190315 08:41:45.455202 207233 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190315 08:41:46.460686 207233 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190315 08:41:47.482410 207233 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190315 08:41:48.499576 207233 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
W190315 08:41:49.294293 208231 storage/store_rebalancer.go:227  [s6,store-rebalancer] StorePool missing descriptor for local store
I190315 08:41:49.528236 217430 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190315 08:41:49.528712 217427 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190315 08:41:49.529108 217427 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
I190315 08:41:49.529447 217428 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190315 08:41:49.529807 217428 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
I190315 08:41:49.530139 217330 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190315 08:41:49.530543 217429 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190315 08:41:49.530923 217431 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
W190315 08:41:49.785552 207480 gossip/gossip.go:1496  [n2] no incoming or outgoing connections
--- FAIL: TestStoreRangeRemoveDead (47.53s)
    <autogenerated>:1: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1]
        goroutine 207233 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc000e7d7a0, 0x4a69580)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4adcc20, 0xc001ac0400, 0xc000e7d770)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc001ac0400)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2944 +0xadb
        testing.tRunner(0xc001ac0400, 0x3da15d0)
        	/usr/local/go/src/testing/testing.go:827 +0x163
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/04ef15974085e14f758b20c552a84052eac9fa2b

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1180936&tab=buildLog

```
I190316 05:37:41.342265 587730 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190316 05:37:42.345330 587730 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190316 05:37:43.436669 587730 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190316 05:37:44.437226 587730 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190316 05:37:45.437653 587730 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190316 05:37:46.442617 587730 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190316 05:37:47.450628 587730 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190316 05:37:48.452189 587730 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190316 05:37:49.459247 587730 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190316 05:37:50.459738 587730 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190316 05:37:51.460086 587730 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190316 05:37:52.460766 587730 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190316 05:37:53.461270 587730 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190316 05:37:54.463157 587730 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190316 05:37:55.471652 587730 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190316 05:37:56.472553 587730 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190316 05:37:57.473140 587730 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190316 05:37:58.475694 587730 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190316 05:37:59.476471 692448 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190316 05:37:59.476654 692448 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190316 05:37:59.476768 692443 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190316 05:37:59.476834 692444 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190316 05:37:59.476856 692445 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190316 05:37:59.476877 692446 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190316 05:37:59.476896 692447 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
--- FAIL: TestStoreRangeRemoveDead (46.12s)
    client_raft_test.go:2944: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1]
        goroutine 587730 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc0018d9020, 0x3409900)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xa7
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x346f8c0, 0xc0007b6000, 0xc0018d8ff0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x103
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc0007b6000)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2944 +0x580
        testing.tRunner(0xc0007b6000, 0x2e290f0)
        	/usr/local/go/src/testing/testing.go:827 +0xbf
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x35c

```

SHA: https://github.com/cockroachdb/cockroach/commits/04ef15974085e14f758b20c552a84052eac9fa2b

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1181234&tab=buildLog

```
I190316 06:21:26.646427 215724 gossip/gossip.go:1510  [n6] node has connected to cluster via gossip
I190316 06:21:27.605468 214656 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
W190316 06:21:28.446783 215724 gossip/gossip.go:1501  [n6] first range unavailable; trying remaining resolvers
I190316 06:21:28.474883 223851 gossip/client.go:128  [n6] started gossip client to 127.0.0.1:41393
I190316 06:21:28.631571 214656 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
W190316 06:21:29.186125 215021 gossip/gossip.go:1501  [n3] first range unavailable; trying remaining resolvers
I190316 06:21:29.268544 224054 gossip/client.go:128  [n3] started gossip client to 127.0.0.1:35463
I190316 06:21:29.390939 215420 gossip/client.go:133  [n5] closing client to n3 (127.0.0.1:42479): stopping outgoing client to n3 (127.0.0.1:42479); already have incoming
I190316 06:21:29.648254 214656 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190316 06:21:30.385572 215724 gossip/gossip.go:1510  [n6] node has connected to cluster via gossip
I190316 06:21:30.660586 214656 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190316 06:21:31.640325 215021 gossip/gossip.go:1510  [n3] node has connected to cluster via gossip
I190316 06:21:31.674436 214656 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190316 06:21:32.720088 224593 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190316 06:21:32.721700 224588 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190316 06:21:32.734611 224593 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190316 06:21:32.736834 224589 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190316 06:21:32.737217 224589 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
I190316 06:21:32.737513 224590 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190316 06:21:32.737827 224592 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190316 06:21:32.738247 224591 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190316 06:21:32.739382 224591 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
I190316 06:21:32.740446 224590 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
I190316 06:21:33.093702 222680 gossip/client.go:133  [n3] closing client to n4 (127.0.0.1:33465): rpc error: code = Internal desc = grpc: compressed flag set with identity or empty encoding
--- FAIL: TestStoreRangeRemoveDead (47.68s)
    <autogenerated>:1: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1]
        goroutine 214656 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc001522d20, 0x4a699c0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4add060, 0xc001830000, 0xc001522cf0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc001830000)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2944 +0xadb
        testing.tRunner(0xc001830000, 0x3da19e0)
        	/usr/local/go/src/testing/testing.go:827 +0x163
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/c32bec552dd89861fbc899a662c1c0cb78acc921

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1182336&tab=buildLog

```
I190317 06:11:33.090098 217386 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190317 06:11:34.164190 217386 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190317 06:11:35.180584 227267 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190317 06:11:35.181103 227263 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190317 06:11:35.181400 227263 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190317 06:11:35.181676 227262 util/stop/stopper.go:546  quiescing; tasks left:
1      mtc send
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190317 06:11:35.195651 227265 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190317 06:11:35.199185 227264 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190317 06:11:35.203777 227264 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190317 06:11:35.204891 227266 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190317 06:11:35.207886 227266 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
I190317 06:11:35.212022 227262 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
W190317 06:11:35.216142 217508 storage/store.go:1485  [s1,r1/1:/M{in-ax}] could not gossip first range descriptor: node unavailable; try another peer
I190317 06:11:35.229300 227262 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
W190317 06:11:35.232486 217921 internal/client/txn.go:509  [hb] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (server shutdown)
I190317 06:11:35.232800 217921 storage/node_liveness.go:775  [hb] retrying liveness update after storage.errRetryLiveness: result is ambiguous (server shutdown)
W190317 06:11:35.233225 217586 internal/client/txn.go:509  [hb] failure aborting transaction: node unavailable; try another peer; abort caused by: failed to send RPC: sending to all 1 replicas failed; last error: (err: node unavailable; try another peer) <nil>
W190317 06:11:35.233745 217586 storage/node_liveness.go:458  [hb] failed node liveness heartbeat: failed to send RPC: sending to all 1 replicas failed; last error: (err: node unavailable; try another peer) <nil>
W190317 06:11:35.234855 217921 internal/client/txn.go:509  [hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W190317 06:11:35.235437 217921 storage/node_liveness.go:458  [hb] failed node liveness heartbeat: node unavailable; try another peer
W190317 06:11:35.469144 217585 gossip/gossip.go:1496  [n3] no incoming or outgoing connections
--- FAIL: TestStoreRangeRemoveDead (46.50s)
    <autogenerated>:1: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1]
        goroutine 217386 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc001464180, 0x4a69b60)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4add200, 0xc001b18700, 0xc001464150)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc001b18700)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2944 +0xadb
        testing.tRunner(0xc001b18700, 0x3da18a0)
        	/usr/local/go/src/testing/testing.go:827 +0x163
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/c2939ec9a4f15b7fb8683a5805deeb241953e7aa

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1183472&tab=buildLog

```
I190318 07:22:48.318325 210933 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190318 07:22:49.354567 210933 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190318 07:22:50.383047 210933 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190318 07:22:51.390550 210933 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190318 07:22:52.449694 210933 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190318 07:22:53.455399 210933 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190318 07:22:54.465354 210933 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190318 07:22:55.478755 210933 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190318 07:22:56.498427 210933 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190318 07:22:57.517790 210933 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190318 07:22:58.529021 210933 storage/client_raft_test.go:2944  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190318 07:22:59.532798 221024 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190318 07:22:59.533173 221026 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190318 07:22:59.534714 221023 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190318 07:22:59.534979 221026 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
I190318 07:22:59.535606 221023 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190318 07:22:59.535973 221022 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190318 07:22:59.533502 221021 util/stop/stopper.go:546  quiescing; tasks left:
1      mtc send
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190318 07:22:59.536979 221021 util/stop/stopper.go:546  quiescing; tasks left:
1      mtc send
1      [async] intent_resolver_gc_batcher
I190318 07:22:59.532936 221025 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190318 07:22:59.539872 221025 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
W190318 07:22:59.542076 210998 storage/store.go:1485  [s1,r1/1:/M{in-ax}] could not gossip first range descriptor: node unavailable; try another peer
I190318 07:22:59.544967 221021 util/stop/stopper.go:546  quiescing; tasks left:
1      mtc send
I190318 07:22:59.551876 221022 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
W190318 07:22:59.788085 211104 gossip/gossip.go:1496  [n2] no incoming or outgoing connections
--- FAIL: TestStoreRangeRemoveDead (47.35s)
    <autogenerated>:1: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1]
        goroutine 210933 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc0014e0660, 0x4a759a0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4ae90c0, 0xc000cf8a00, 0xc0014e0600)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc000cf8a00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2944 +0xadb
        testing.tRunner(0xc000cf8a00, 0x3dab860)
        	/usr/local/go/src/testing/testing.go:827 +0x163
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/f95d45653df6be5587fb9887de241f50b6932000

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1186381&tab=buildLog

```
I190319 08:11:52.700100 209428 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190319 08:11:53.750809 209428 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190319 08:11:54.774251 209428 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190319 08:11:55.794053 209428 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190319 08:11:56.801225 209428 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190319 08:11:57.842549 209428 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190319 08:11:58.907709 209428 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190319 08:11:59.948076 209428 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190319 08:12:00.964283 209428 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190319 08:12:01.997288 209428 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190319 08:12:03.024224 209428 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190319 08:12:04.044513 209428 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190319 08:12:05.051398 209428 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
W190319 08:12:05.986863 210404 storage/store_rebalancer.go:227  [s6,store-rebalancer] StorePool missing descriptor for local store
I190319 08:12:06.068974 209428 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190319 08:12:07.079494 219728 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190319 08:12:07.085411 219725 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190319 08:12:07.087388 219725 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190319 08:12:07.089170 219726 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190319 08:12:07.091021 219726 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190319 08:12:07.091717 219727 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190319 08:12:07.092944 219727 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190319 08:12:07.099071 219724 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190319 08:12:07.099588 219729 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
W190319 08:12:07.356073 209392 gossip/gossip.go:1496  [n2] no incoming or outgoing connections
--- FAIL: TestStoreRangeRemoveDead (47.44s)
    <autogenerated>:1: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1]
        goroutine 209428 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc000bbc5a0, 0x4a7e140)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4af1980, 0xc001793200, 0xc000bbc570)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc001793200)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2940 +0xadb
        testing.tRunner(0xc001793200, 0x3db5a18)
        	/usr/local/go/src/testing/testing.go:827 +0x163
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/3a7ea2d8c9d4a3e0d97f8f106fcf95b3f03765ec

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1187991&tab=buildLog

```
I190320 06:02:45.299285 204344 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190320 06:02:46.317231 204344 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190320 06:02:47.334260 204344 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190320 06:02:48.396506 204344 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190320 06:02:49.464594 204344 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190320 06:02:50.472199 204344 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190320 06:02:51.490739 204344 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190320 06:02:52.548441 204344 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190320 06:02:53.582836 204344 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190320 06:02:54.627074 204344 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190320 06:02:55.690301 204344 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190320 06:02:56.706404 204344 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190320 06:02:57.760007 204344 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190320 06:02:58.777915 204344 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190320 06:02:59.797468 204344 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190320 06:03:00.921818 204344 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190320 06:03:01.933139 213373 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190320 06:03:01.933978 213369 util/stop/stopper.go:546  quiescing; tasks left:
1      mtc send
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190320 06:03:01.936218 213369 util/stop/stopper.go:546  quiescing; tasks left:
1      mtc send
1      [async] intent_resolver_gc_batcher
I190320 06:03:01.937748 213371 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190320 06:03:01.941981 213372 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190320 06:03:01.944854 213370 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190320 06:03:01.944986 213373 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190320 06:03:01.945269 213374 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190320 06:03:01.945739 213369 util/stop/stopper.go:546  quiescing; tasks left:
1      mtc send
--- FAIL: TestStoreRangeRemoveDead (47.40s)
    <autogenerated>:1: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1]
        goroutine 204344 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc008077200, 0x4a85820)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4af9100, 0xc001866500, 0xc0080771d0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc001866500)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2940 +0xadb
        testing.tRunner(0xc001866500, 0x3dbb478)
        	/usr/local/go/src/testing/testing.go:827 +0x163
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/dfa23c01e4ea39b19ca8b2e5c8a4e7cf9b9445f4

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1190951&tab=buildLog

```
I190321 07:12:09.925635 212467 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190321 07:12:10.965018 212467 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190321 07:12:11.993649 212467 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190321 07:12:13.002179 212467 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190321 07:12:14.009749 212467 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190321 07:12:15.064847 212467 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190321 07:12:16.106500 212467 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190321 07:12:17.161217 212467 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190321 07:12:18.168019 212467 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190321 07:12:19.196182 212467 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190321 07:12:20.221806 212467 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190321 07:12:21.225015 212467 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190321 07:12:22.238306 212467 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190321 07:12:23.265259 223211 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190321 07:12:23.265703 223206 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190321 07:12:23.265994 223206 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
I190321 07:12:23.266252 223208 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190321 07:12:23.266568 223208 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190321 07:12:23.266794 223209 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190321 07:12:23.267073 223209 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190321 07:12:23.267382 223210 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190321 07:12:23.267687 223210 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190321 07:12:23.267974 223207 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190321 07:12:23.275771 223207 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
W190321 07:12:23.548571 212564 gossip/gossip.go:1496  [n2] no incoming or outgoing connections
--- FAIL: TestStoreRangeRemoveDead (47.24s)
    <autogenerated>:1: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1]
        goroutine 212467 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc00b7deba0, 0x4a87e80)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4afb700, 0xc000872400, 0xc00b7deb70)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc000872400)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2940 +0xadb
        testing.tRunner(0xc000872400, 0x3dbd5d0)
        	/usr/local/go/src/testing/testing.go:827 +0x163
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/a200cea4368ec90aaee12337d7ab5f9ca555108f

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1193444&tab=buildLog

```
I190322 07:47:40.068521 206263 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190322 07:47:41.137602 206263 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190322 07:47:42.166102 206263 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190322 07:47:43.177377 206263 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190322 07:47:44.203648 206263 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190322 07:47:45.209893 206263 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190322 07:47:46.226409 206263 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190322 07:47:47.277093 206263 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190322 07:47:48.354767 206263 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190322 07:47:49.371625 206263 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190322 07:47:50.376789 206263 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190322 07:47:51.429157 216119 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190322 07:47:51.429656 216119 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
I190322 07:47:51.430776 216116 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190322 07:47:51.435475 216117 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190322 07:47:51.437977 216115 util/stop/stopper.go:546  quiescing; tasks left:
2      mtc send
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190322 07:47:51.440993 216118 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190322 07:47:51.448483 216120 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190322 07:47:51.451604 216116 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
I190322 07:47:51.459376 216115 util/stop/stopper.go:546  quiescing; tasks left:
1      mtc send
W190322 07:47:51.501960 206869 internal/client/txn.go:509  [hb] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (server shutdown)
I190322 07:47:51.502235 206869 storage/node_liveness.go:775  [hb] retrying liveness update after storage.errRetryLiveness: result is ambiguous (server shutdown)
W190322 07:47:51.507526 206869 internal/client/txn.go:509  [hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W190322 07:47:51.507968 206869 storage/node_liveness.go:458  [hb] failed node liveness heartbeat: node unavailable; try another peer
--- FAIL: TestStoreRangeRemoveDead (46.85s)
    <autogenerated>:1: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1]
        goroutine 206263 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc000954930, 0x4a8a6c0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4afdee0, 0xc001684b00, 0xc000954900)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc001684b00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2940 +0xadb
        testing.tRunner(0xc001684b00, 0x3dbf370)
        	/usr/local/go/src/testing/testing.go:827 +0x163
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/b5768aecd39461ab9a54e2e7db059a3fe8b00459

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1193402&tab=buildLog

```
I190322 08:53:26.946585 211312 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190322 08:53:27.953326 211312 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190322 08:53:28.983101 211312 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190322 08:53:30.087755 211312 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190322 08:53:31.101630 211312 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190322 08:53:32.121997 211312 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190322 08:53:33.185568 211312 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190322 08:53:34.215176 211312 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190322 08:53:35.263082 211312 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190322 08:53:36.275167 211312 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190322 08:53:37.346481 211312 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190322 08:53:38.391226 211312 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190322 08:53:39.420500 211312 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190322 08:53:40.472945 211312 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190322 08:53:41.478080 220810 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190322 08:53:41.478423 220805 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190322 08:53:41.478770 220806 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190322 08:53:41.479026 220806 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190322 08:53:41.479242 220807 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190322 08:53:41.479518 220807 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190322 08:53:41.479719 220808 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190322 08:53:41.480106 220808 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
I190322 08:53:41.479520 220809 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
W190322 08:53:41.653631 212391 gossip/gossip.go:1496  [n6] no incoming or outgoing connections
W190322 08:53:41.654119 211619 gossip/gossip.go:1496  [n3] no incoming or outgoing connections
--- FAIL: TestStoreRangeRemoveDead (46.54s)
    <autogenerated>:1: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1]
        goroutine 211312 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc000af9bf0, 0x4a85600)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4af8e80, 0xc001414800, 0xc000af9bc0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc001414800)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2940 +0xadb
        testing.tRunner(0xc001414800, 0x3dbb6a0)
        	/usr/local/go/src/testing/testing.go:827 +0x163
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/6cac063ae1cb578130afbafb2abf4035268a10c9

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1194679&tab=buildLog

```
I190323 05:46:22.774341 661487 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 05:46:23.780878 661487 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 05:46:24.781327 661487 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 05:46:25.781940 661487 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 05:46:26.810279 661487 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 05:46:27.813236 661487 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 05:46:28.814188 661487 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 05:46:29.815012 661487 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 05:46:30.815675 661487 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 05:46:31.816256 661487 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 05:46:32.816707 661487 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 05:46:33.818709 661487 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 05:46:34.819367 661487 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 05:46:35.820190 865128 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190323 05:46:35.820194 865124 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190323 05:46:35.820249 865125 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190323 05:46:35.820263 865125 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190323 05:46:35.820193 865027 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190323 05:46:35.820276 865128 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190323 05:46:35.820285 865027 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190323 05:46:35.820321 865126 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190323 05:46:35.820338 865127 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190323 05:46:35.820345 865126 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
I190323 05:46:35.820363 865127 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
W190323 05:46:35.822134 663080 gossip/gossip.go:1496  [n6] no incoming or outgoing connections
--- FAIL: TestStoreRangeRemoveDead (45.53s)
    client_raft_test.go:2940: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1]
        goroutine 661487 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc000ddc180, 0x3422160)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xa7
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x34881c0, 0xc000dd0800, 0xc000ddc150)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x103
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc000dd0800)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2940 +0x580
        testing.tRunner(0xc000dd0800, 0x2e41410)
        	/usr/local/go/src/testing/testing.go:827 +0xbf
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x35c

```

SHA: https://github.com/cockroachdb/cockroach/commits/6cac063ae1cb578130afbafb2abf4035268a10c9

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1195257&tab=buildLog

```
I190323 06:23:54.927671 207765 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 06:23:55.952685 207765 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 06:23:56.971687 207765 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 06:23:57.977291 207765 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 06:23:58.989624 207765 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 06:24:00.041062 207765 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 06:24:01.122892 207765 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 06:24:02.170536 207765 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 06:24:03.185468 207765 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 06:24:04.214848 207765 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 06:24:05.245425 207765 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 06:24:06.292272 207765 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 06:24:07.307763 207765 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 06:24:08.323020 207765 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 06:24:09.362220 207765 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 06:24:10.376328 207765 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 06:24:11.452298 207765 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 06:24:12.509650 207765 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 06:24:13.513279 217884 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190323 06:24:13.513702 217882 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190323 06:24:13.514281 217883 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190323 06:24:13.514677 217885 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190323 06:24:13.515067 217886 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190323 06:24:13.519796 217887 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190323 06:24:13.521521 217883 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
--- FAIL: TestStoreRangeRemoveDead (47.48s)
    <autogenerated>:1: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1]
        goroutine 207765 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc0173b12c0, 0x4a8ad00)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4afe520, 0xc0009bcf00, 0xc0173b1290)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc0009bcf00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2940 +0xadb
        testing.tRunner(0xc0009bcf00, 0x3dbf938)
        	/usr/local/go/src/testing/testing.go:827 +0x163
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/9399d559ae196e5cf2ad122195048ff9115ab56a

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1195316&tab=buildLog

```
I190323 08:52:14.460369 204624 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 08:52:15.528574 204624 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 08:52:16.548130 204624 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 08:52:17.611219 204624 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 08:52:18.651669 204624 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 08:52:19.705269 204624 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 08:52:20.749877 204624 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 08:52:21.762192 204624 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 08:52:22.839215 204624 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 08:52:23.871243 204624 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 08:52:24.916587 204624 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 08:52:25.926475 204624 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 08:52:26.942299 204624 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 08:52:27.979739 204624 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 08:52:28.988620 204624 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190323 08:52:30.017571 214081 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190323 08:52:30.019662 214077 util/stop/stopper.go:546  quiescing; tasks left:
1      mtc send
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190323 08:52:30.020042 214078 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190323 08:52:30.020378 214079 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190323 08:52:30.020664 214080 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190323 08:52:30.021005 214080 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
I190323 08:52:30.021239 214082 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190323 08:52:30.042912 214081 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
W190323 08:52:30.063781 204811 storage/store.go:1485  [s1,r1/1:/M{in-ax}] could not gossip first range descriptor: node unavailable; try another peer
W190323 08:52:30.380605 205494 gossip/gossip.go:1501  [n5] first range unavailable; trying remaining resolvers
--- FAIL: TestStoreRangeRemoveDead (46.50s)
    <autogenerated>:1: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1]
        goroutine 204624 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc000b95110, 0x4a855a0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4af8e20, 0xc0005dd300, 0xc000b950e0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc0005dd300)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2940 +0xadb
        testing.tRunner(0xc0005dd300, 0x3dbb698)
        	/usr/local/go/src/testing/testing.go:827 +0x163
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/5a746073c3f8ede851f37dd895cf1a91d6dcc3cf

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1196647&tab=buildLog

```
I190324 06:33:03.861643 208525 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190324 06:33:04.871640 208525 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190324 06:33:05.891934 208525 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190324 06:33:06.894557 208525 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190324 06:33:07.919295 208525 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190324 06:33:08.930144 208525 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190324 06:33:09.947401 208525 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190324 06:33:10.968156 208525 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190324 06:33:12.021632 208525 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190324 06:33:13.030140 208525 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190324 06:33:14.077020 208525 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190324 06:33:15.114535 208525 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190324 06:33:16.168322 208525 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190324 06:33:17.196607 208525 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190324 06:33:18.208358 218760 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190324 06:33:18.208824 218756 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190324 06:33:18.209268 218758 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190324 06:33:18.209592 218757 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190324 06:33:18.209979 218759 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190324 06:33:18.212569 218759 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190324 06:33:18.212808 218761 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190324 06:33:18.235386 218756 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
W190324 06:33:18.257421 208712 storage/store.go:1487  [s1,r1/1:/M{in-ax}] could not gossip first range descriptor: node unavailable; try another peer
I190324 06:33:18.258234 218761 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
W190324 06:33:18.455718 208733 gossip/gossip.go:1496  [n2] no incoming or outgoing connections
--- FAIL: TestStoreRangeRemoveDead (47.51s)
    <autogenerated>:1: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1]
        goroutine 208525 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc001521d40, 0x4a8b320)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4afeb40, 0xc000c82000, 0xc001521d10)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc000c82000)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2940 +0xadb
        testing.tRunner(0xc000c82000, 0x3dbfe18)
        	/usr/local/go/src/testing/testing.go:827 +0x163
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/c59f5347d5424edb90575fb0fd50bad677953752

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1196732&tab=buildLog

```
I190324 06:48:45.598374 206279 gossip/gossip.go:1510  [n6] node has connected to cluster via gossip
I190324 06:48:46.073294 205675 storage/node_liveness.go:775  [hb] retrying liveness update after storage.errRetryLiveness: result is ambiguous (context deadline exceeded)
W190324 06:48:46.083253 205675 storage/node_liveness.go:458  [hb] failed node liveness heartbeat: operation "node liveness heartbeat" timed out after 450ms
I190324 06:48:46.396289 205592 gossip/gossip.go:1510  [n2] node has connected to cluster via gossip
I190324 06:48:46.576340 205249 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190324 06:48:47.032237 205675 storage/node_liveness.go:775  [hb] retrying liveness update after storage.errRetryLiveness: result is ambiguous (context deadline exceeded)
W190324 06:48:47.032842 205675 storage/node_liveness.go:458  [hb] failed node liveness heartbeat: operation "node liveness heartbeat" timed out after 450ms
I190324 06:48:47.647067 205249 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190324 06:48:47.977956 205675 storage/node_liveness.go:775  [hb] retrying liveness update after storage.errRetryLiveness: result is ambiguous (context deadline exceeded)
W190324 06:48:47.979070 205675 storage/node_liveness.go:458  [hb] failed node liveness heartbeat: operation "node liveness heartbeat" timed out after 450ms
I190324 06:48:48.684383 205249 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190324 06:48:49.221764 205675 storage/node_liveness.go:775  [hb] retrying liveness update after storage.errRetryLiveness: result is ambiguous (context deadline exceeded)
W190324 06:48:49.222362 205675 storage/node_liveness.go:458  [hb] failed node liveness heartbeat: operation "node liveness heartbeat" timed out after 450ms
I190324 06:48:49.688851 205249 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190324 06:48:50.696175 213660 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190324 06:48:50.696777 213657 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190324 06:48:50.697458 213659 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190324 06:48:50.698058 213658 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190324 06:48:50.698509 213661 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190324 06:48:50.698828 213661 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
I190324 06:48:50.699111 213662 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
W190324 06:48:51.245389 206279 gossip/gossip.go:1501  [n6] first range unavailable; trying remaining resolvers
W190324 06:48:51.246357 205879 gossip/gossip.go:1496  [n4] no incoming or outgoing connections
--- FAIL: TestStoreRangeRemoveDead (46.87s)
    <autogenerated>:1: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1]
        goroutine 205249 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc001a5f710, 0x4a85c00)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4af9480, 0xc000b1ee00, 0xc001a5f6b0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc000b1ee00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2940 +0xadb
        testing.tRunner(0xc000b1ee00, 0x3dbbbb8)
        	/usr/local/go/src/testing/testing.go:827 +0x163
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/7bc9ea5fbe0c0082fdcfd408245a79c62b00edd4

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1197973&tab=buildLog

```
I190325 06:18:36.302661 209117 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190325 06:18:37.321495 209117 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190325 06:18:38.386188 209117 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190325 06:18:39.401115 209117 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190325 06:18:40.427018 209117 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190325 06:18:41.438105 209117 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190325 06:18:42.453979 209117 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190325 06:18:43.462267 209117 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190325 06:18:44.477086 219781 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190325 06:18:44.477752 219699 util/stop/stopper.go:546  quiescing; tasks left:
1      mtc send
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190325 06:18:44.477817 219781 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190325 06:18:44.478248 219700 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190325 06:18:44.496344 219782 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190325 06:18:44.499582 219783 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190325 06:18:44.500706 219784 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190325 06:18:44.502190 219700 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
W190325 06:18:44.520864 209576 internal/client/txn.go:509  [hb] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (server shutdown)
I190325 06:18:44.521244 209576 storage/node_liveness.go:775  [hb] retrying liveness update after storage.errRetryLiveness: result is ambiguous (server shutdown)
W190325 06:18:44.523662 209576 internal/client/txn.go:509  [hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W190325 06:18:44.524110 209576 storage/node_liveness.go:458  [hb] failed node liveness heartbeat: node unavailable; try another peer
W190325 06:18:44.533629 209288 storage/store.go:1487  [s1,r1/1:/M{in-ax}] could not gossip first range descriptor: node unavailable; try another peer
W190325 06:18:44.746731 210139 gossip/gossip.go:1496  [n6] no incoming or outgoing connections
W190325 06:18:44.748585 209575 gossip/gossip.go:1496  [n3] no incoming or outgoing connections
--- FAIL: TestStoreRangeRemoveDead (46.91s)
    <autogenerated>:1: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1]
        goroutine 209117 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc000f53800, 0x4a8eb40)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4b02400, 0xc000c64500, 0xc000f537d0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc000c64500)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2940 +0xadb
        testing.tRunner(0xc000c64500, 0x3dc2ea8)
        	/usr/local/go/src/testing/testing.go:827 +0x163
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/ec89b45cea7e8a6dd92a9bfd60e0cc06842e06d8

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1198112&tab=buildLog

```
1      [async] intent_resolver_gc_batcher
I190325 06:39:55.425318 223325 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190325 06:39:55.429188 223326 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190325 06:39:55.430101 223324 util/stop/stopper.go:546  quiescing; tasks left:
5      mtc send
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190325 06:39:55.430424 223327 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190325 06:39:55.430981 223326 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
W190325 06:39:55.434283 213678 internal/client/txn.go:509  [hb] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (server shutdown)
I190325 06:39:55.434522 213678 storage/node_liveness.go:775  [hb] retrying liveness update after storage.errRetryLiveness: result is ambiguous (server shutdown)
W190325 06:39:55.437118 213783 storage/store.go:1485  [s1,r1/1:/M{in-ax}] could not gossip first range descriptor: node unavailable; try another peer
W190325 06:39:55.437668 213678 internal/client/txn.go:509  [hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W190325 06:39:55.438263 213678 storage/node_liveness.go:458  [hb] failed node liveness heartbeat: node unavailable; try another peer
I190325 06:39:55.439330 223324 util/stop/stopper.go:546  quiescing; tasks left:
2      mtc send
I190325 06:39:55.443429 223325 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
W190325 06:39:55.457283 214620 internal/client/txn.go:509  [hb] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (server shutdown)
I190325 06:39:55.457553 214620 storage/node_liveness.go:775  [hb] retrying liveness update after storage.errRetryLiveness: result is ambiguous (server shutdown)
W190325 06:39:55.459216 214620 internal/client/txn.go:509  [hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W190325 06:39:55.460113 214620 storage/node_liveness.go:458  [hb] failed node liveness heartbeat: node unavailable; try another peer
W190325 06:39:55.462229 214044 internal/client/txn.go:509  [hb] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (server shutdown)
I190325 06:39:55.463848 223324 util/stop/stopper.go:546  quiescing; tasks left:
1      mtc send
I190325 06:39:55.466379 214044 storage/node_liveness.go:775  [hb] retrying liveness update after storage.errRetryLiveness: result is ambiguous (server shutdown)
W190325 06:39:55.468537 214044 internal/client/txn.go:509  [hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W190325 06:39:55.469077 214044 storage/node_liveness.go:458  [hb] failed node liveness heartbeat: node unavailable; try another peer
--- FAIL: TestStoreRangeRemoveDead (46.99s)
    <autogenerated>:1: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1]
        goroutine 213687 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc000932060, 0x4a89420)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4afcd40, 0xc000c69900, 0xc000932030)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc000c69900)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2940 +0xadb
        testing.tRunner(0xc000c69900, 0x3dbec48)
        	/usr/local/go/src/testing/testing.go:827 +0x163
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/25398c010b2af75b11fed189680ea6b9645f0cf5

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1200016&tab=buildLog

```
I190326 05:52:18.695576 631050 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190326 05:52:19.696377 631050 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190326 05:52:20.697152 631050 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190326 05:52:21.697801 631050 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190326 05:52:22.698979 631050 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190326 05:52:23.700705 631050 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190326 05:52:24.704873 631050 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190326 05:52:25.716743 631050 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190326 05:52:26.717144 631050 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190326 05:52:27.718902 631050 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190326 05:52:28.719404 631050 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190326 05:52:29.720400 631050 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190326 05:52:30.720870 631050 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190326 05:52:31.725009 777361 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190326 05:52:31.725046 777361 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190326 05:52:31.725070 777356 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190326 05:52:31.725086 777356 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190326 05:52:31.725104 777357 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190326 05:52:31.725122 777357 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190326 05:52:31.725146 777358 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190326 05:52:31.725163 777358 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190326 05:52:31.725199 777359 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190326 05:52:31.725213 777360 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190326 05:52:31.725264 777359 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
W190326 05:52:31.727267 631132 gossip/gossip.go:1496  [n2] no incoming or outgoing connections
--- FAIL: TestStoreRangeRemoveDead (45.58s)
    client_raft_test.go:2940: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1]
        goroutine 631050 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc001e2df50, 0x34262c0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xa7
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x348c4e0, 0xc001182000, 0xc001e2df20)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x103
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc001182000)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2940 +0x580
        testing.tRunner(0xc001182000, 0x2e41d08)
        	/usr/local/go/src/testing/testing.go:827 +0xbf
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x35c

```

SHA: https://github.com/cockroachdb/cockroach/commits/7f8a0969e8e9eb7e9fc0d2fe96e03849d30dd561

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1200641&tab=buildLog

```
I190326 06:29:47.101070 207553 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190326 06:29:48.108112 207553 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190326 06:29:49.130894 207553 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190326 06:29:50.135159 207553 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190326 06:29:51.141655 207553 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190326 06:29:52.146729 207553 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190326 06:29:53.157256 207553 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190326 06:29:54.172631 207553 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190326 06:29:55.180881 207553 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190326 06:29:56.201733 207553 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190326 06:29:57.207256 207553 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190326 06:29:58.215319 207553 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190326 06:29:59.301036 216677 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190326 06:29:59.307020 216678 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190326 06:29:59.311939 216676 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190326 06:29:59.312129 216675 util/stop/stopper.go:546  quiescing; tasks left:
1      mtc send
1      [async] storage.pendingLeaseRequest: requesting lease
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190326 06:29:59.313419 216676 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
I190326 06:29:59.319052 216680 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190326 06:29:59.319967 216679 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190326 06:29:59.320089 216680 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
W190326 06:29:59.323180 207772 storage/store.go:1485  [s1,r1/1:/M{in-ax}] could not gossip first range descriptor: node unavailable; try another peer
I190326 06:29:59.361544 216675 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] storage.pendingLeaseRequest: requesting lease
W190326 06:29:59.523120 208099 gossip/gossip.go:1501  [n3] first range unavailable; trying remaining resolvers
I190326 06:29:59.536069 216652 gossip/client.go:128  [n3] started gossip client to 127.0.0.1:46797
--- FAIL: TestStoreRangeRemoveDead (47.10s)
    <autogenerated>:1: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1]
        goroutine 207553 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc00147d470, 0x4a93c80)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4b07600, 0xc000ffc400, 0xc00147d440)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc000ffc400)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2940 +0xadb
        testing.tRunner(0xc000ffc400, 0x3dbe6d8)
        	/usr/local/go/src/testing/testing.go:827 +0x163
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/25398c010b2af75b11fed189680ea6b9645f0cf5

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1200658&tab=buildLog

```
I190326 09:29:55.286988 202202 gossip/gossip.go:1510  [n6] node has connected to cluster via gossip
I190326 09:29:55.288798 208043 gossip/client.go:128  [n6] started gossip client to 127.0.0.1:35387
I190326 09:29:55.612986 201075 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190326 09:29:56.649225 201075 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190326 09:29:57.669462 201075 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190326 09:29:58.701224 201075 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190326 09:29:59.706037 201075 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190326 09:30:00.721786 201075 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190326 09:30:01.725147 201075 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190326 09:30:02.733100 201075 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190326 09:30:03.746967 201075 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190326 09:30:04.763022 201075 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190326 09:30:05.811030 201075 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190326 09:30:06.819725 201075 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190326 09:30:07.826592 201075 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190326 09:30:08.832976 201075 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190326 09:30:09.898889 201075 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190326 09:30:10.914430 211201 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190326 09:30:10.920323 211196 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190326 09:30:10.923307 211197 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190326 09:30:10.926301 211198 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190326 09:30:10.926668 211200 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190326 09:30:10.927741 211199 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
W190326 09:30:10.933030 201276 storage/store.go:1487  [s1,r1/1:/M{in-ax}] could not gossip first range descriptor: node unavailable; try another peer
W190326 09:30:11.244201 201424 gossip/gossip.go:1496  [n3] no incoming or outgoing connections
--- FAIL: TestStoreRangeRemoveDead (46.79s)
    <autogenerated>:1: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1]
        goroutine 201075 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc000fcb3e0, 0x4a9b9a0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4b0f380, 0xc000e6e000, 0xc000fcb3b0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc000e6e000)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2940 +0xadb
        testing.tRunner(0xc000e6e000, 0x3dc4870)
        	/usr/local/go/src/testing/testing.go:827 +0x163
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/ed21f6da64b022a3e1e550fad5850fdffe2a7d17

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1202808&tab=buildLog

```
I190327 07:27:38.797827 204665 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190327 07:27:39.802910 204665 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190327 07:27:40.822472 204665 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190327 07:27:41.838237 204665 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190327 07:27:42.844286 204665 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190327 07:27:43.867415 204665 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190327 07:27:44.909121 204665 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190327 07:27:45.917343 204665 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190327 07:27:46.920460 204665 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190327 07:27:47.927480 204665 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190327 07:27:48.930976 204665 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190327 07:27:49.933084 204665 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190327 07:27:50.940948 204665 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190327 07:27:51.944322 204665 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190327 07:27:52.958406 215584 util/stop/stopper.go:546  quiescing; tasks left:
1      mtc send
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190327 07:27:52.962501 215584 util/stop/stopper.go:546  quiescing; tasks left:
1      mtc send
I190327 07:27:52.963567 215586 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190327 07:27:52.965383 215585 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190327 07:27:52.965817 215585 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
I190327 07:27:52.966054 215651 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190327 07:27:52.967128 215652 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190327 07:27:52.969704 215652 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190327 07:27:52.970216 215653 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190327 07:27:52.970487 215653 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
W190327 07:27:52.998221 204756 storage/store.go:1498  [s1,r1/1:/M{in-ax}] could not gossip first range descriptor: node unavailable; try another peer
--- FAIL: TestStoreRangeRemoveDead (47.30s)
    <autogenerated>:1: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1]
        goroutine 204665 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc00126bad0, 0x4ad70a0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4b4b9c0, 0xc001934b00, 0xc00126baa0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc001934b00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2940 +0xadb
        testing.tRunner(0xc001934b00, 0x3df15a8)
        	/usr/local/go/src/testing/testing.go:827 +0x163
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/3aadd20bbf0940ef65f8b2cdcda498401ba5d9c6

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1202961&tab=buildLog

```
I190327 09:04:34.923157 203307 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190327 09:04:35.925125 203307 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190327 09:04:36.934490 203307 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190327 09:04:37.948158 203307 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190327 09:04:38.951005 203307 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190327 09:04:39.979336 203307 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190327 09:04:41.016472 203307 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190327 09:04:42.030184 203307 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190327 09:04:43.033478 203307 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190327 09:04:44.036984 203307 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190327 09:04:45.049887 203307 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190327 09:04:46.056266 203307 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190327 09:04:47.084979 203307 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190327 09:04:48.089048 203307 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190327 09:04:49.096913 214807 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190327 09:04:49.097332 214807 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190327 09:04:49.097601 214805 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190327 09:04:49.097838 214805 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190327 09:04:49.098042 214806 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190327 09:04:49.098410 214808 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190327 09:04:49.098656 214808 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
I190327 09:04:49.098911 214809 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190327 09:04:49.099245 214810 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
W190327 09:04:49.318985 204032 gossip/gossip.go:1496  [n5] no incoming or outgoing connections
W190327 09:04:49.319931 203912 gossip/gossip.go:1496  [n4] no incoming or outgoing connections
--- FAIL: TestStoreRangeRemoveDead (46.23s)
    <autogenerated>:1: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1]
        goroutine 203307 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc001263dd0, 0x4a93ae0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4b07460, 0xc00b3f2000, 0xc001263da0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc00b3f2000)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2940 +0xadb
        testing.tRunner(0xc00b3f2000, 0x3dbe558)
        	/usr/local/go/src/testing/testing.go:827 +0x163
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/17565100d1e7c66341e6db3e39bb66202958cb81

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1206073&tab=buildLog

```
I190328 08:03:00.426172 207772 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190328 08:03:01.434997 207772 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190328 08:03:02.443576 207772 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190328 08:03:03.455890 207772 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190328 08:03:04.464642 207772 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190328 08:03:05.469187 207772 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190328 08:03:06.478931 207772 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190328 08:03:07.481318 207772 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190328 08:03:08.484666 207772 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190328 08:03:09.488234 207772 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190328 08:03:10.525138 207772 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190328 08:03:11.547008 207772 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190328 08:03:12.572458 207772 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190328 08:03:13.576353 207772 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190328 08:03:14.593381 218343 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190328 08:03:14.596229 218343 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
I190328 08:03:14.596991 218348 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190328 08:03:14.597473 218347 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190328 08:03:14.598626 218346 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190328 08:03:14.598926 218346 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190328 08:03:14.599151 218344 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190328 08:03:14.599399 218344 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190328 08:03:14.599600 218345 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190328 08:03:14.601146 218347 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190328 08:03:14.624676 218345 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
--- FAIL: TestStoreRangeRemoveDead (47.49s)
    <autogenerated>:1: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1]
        goroutine 207772 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc0018faf60, 0x4aeea00)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4b63460, 0xc001804300, 0xc0018faf30)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc001804300)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2940 +0xaef
        testing.tRunner(0xc001804300, 0x3dfb650)
        	/usr/local/go/src/testing/testing.go:827 +0x163
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/a1e6e9decc9dec15a32bbb6d30efc67ca45a532a

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1206012&tab=buildLog

```
I190328 08:16:16.746020 203793 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190328 08:16:17.755082 203793 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190328 08:16:18.770217 203793 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190328 08:16:19.807510 213078 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190328 08:16:19.812396 213009 util/stop/stopper.go:546  quiescing; tasks left:
2      mtc send
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190328 08:16:19.812906 213009 util/stop/stopper.go:546  quiescing; tasks left:
2      mtc send
1      [async] intent_resolver_gc_batcher
I190328 08:16:19.820218 213011 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190328 08:16:19.820538 213010 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190328 08:16:19.820676 213012 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190328 08:16:19.821048 213077 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190328 08:16:19.821722 213012 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
I190328 08:16:19.822391 213077 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
W190328 08:16:19.825694 203996 storage/store.go:1485  [s1,r1/1:/M{in-ax}] could not gossip first range descriptor: node unavailable; try another peer
I190328 08:16:19.827032 213078 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
I190328 08:16:19.842175 213010 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
I190328 08:16:19.843323 213011 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
I190328 08:16:19.844563 213009 util/stop/stopper.go:546  quiescing; tasks left:
1      mtc send
W190328 08:16:19.854885 204013 internal/client/txn.go:509  [hb] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (server shutdown)
I190328 08:16:19.859966 204013 storage/node_liveness.go:775  [hb] retrying liveness update after storage.errRetryLiveness: result is ambiguous (server shutdown)
W190328 08:16:19.862072 204013 internal/client/txn.go:509  [hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W190328 08:16:19.863511 204013 storage/node_liveness.go:458  [hb] failed node liveness heartbeat: node unavailable; try another peer
W190328 08:16:20.125737 204539 gossip/gossip.go:1496  [n5] no incoming or outgoing connections
W190328 08:16:20.127703 204119 gossip/gossip.go:1496  [n2] no incoming or outgoing connections
--- FAIL: TestStoreRangeRemoveDead (47.25s)
    <autogenerated>:1: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1]
        goroutine 203793 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc000c9ae40, 0x4a92cc0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4b06640, 0xc00153a600, 0xc000c9acf0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc00153a600)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2940 +0xadb
        testing.tRunner(0xc00153a600, 0x3dbe768)
        	/usr/local/go/src/testing/testing.go:827 +0x163
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/d03a34e92d2ee558fb6aedb0709b733a1fab97f4

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1207922&tab=buildLog

```
I190329 05:42:27.274020 380947 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190329 05:42:28.275581 380947 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190329 05:42:29.279701 380947 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190329 05:42:30.296598 380947 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190329 05:42:31.297178 380947 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190329 05:42:32.297762 380947 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190329 05:42:33.298893 380947 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190329 05:42:34.299659 380947 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190329 05:42:35.306162 380947 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190329 05:42:36.307093 380947 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190329 05:42:37.324231 380947 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190329 05:42:38.327982 487827 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190329 05:42:38.328028 487827 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190329 05:42:38.328110 487822 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190329 05:42:38.328137 487822 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190329 05:42:38.328161 487823 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190329 05:42:38.328188 487823 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
I190329 05:42:38.328217 487824 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190329 05:42:38.328244 487824 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190329 05:42:38.328282 487825 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190329 05:42:38.328308 487825 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190329 05:42:38.328353 487826 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190329 05:42:38.328380 487826 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
W190329 05:42:38.334523 382415 gossip/gossip.go:1496  [n6] no incoming or outgoing connections
W190329 05:42:38.334600 381071 gossip/gossip.go:1496  [n2] no incoming or outgoing connections
W190329 05:42:38.334648 381609 gossip/gossip.go:1496  [n4] no incoming or outgoing connections
--- FAIL: TestStoreRangeRemoveDead (45.62s)
    client_raft_test.go:2940: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1]
        goroutine 380947 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc00176e060, 0x3452be0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xa7
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x34b9e20, 0xc000ca4000, 0xc00176e030)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x103
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc000ca4000)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2940 +0x584
        testing.tRunner(0xc000ca4000, 0x2e62e60)
        	/usr/local/go/src/testing/testing.go:827 +0xbf
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x35c

```

SHA: https://github.com/cockroachdb/cockroach/commits/d03a34e92d2ee558fb6aedb0709b733a1fab97f4

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1208530&tab=buildLog

```
W190329 06:26:28.775943 215415 storage/store.go:1497  [s4,r1/2:/M{in-ax}] could not gossip first range descriptor: node unavailable; try another peer
W190329 06:26:28.798421 218217 storage/raft_transport.go:281  unable to accept Raft message from (n4,s4):2: no handler registered for (n2,s2):4
W190329 06:26:28.800669 218283 storage/store.go:3577  [s4,r1/2:/M{in-ax}] raft error: node 2 claims to not contain store 2 for replica (n2,s2):4: store 2 was not found
W190329 06:26:28.801162 218281 storage/raft_transport.go:583  while processing outgoing Raft queue to node 2: store 2 was not found:
W190329 06:26:28.829195 216569 storage/raft_transport.go:281  unable to accept Raft message from (n4,s4):2: no handler registered for (n1,s1):1
W190329 06:26:28.845276 215138 gossip/gossip.go:1501  [n4] first range unavailable; trying remaining resolvers
I190329 06:26:28.853848 222838 gossip/client.go:128  [n4] started gossip client to 127.0.0.1:40657
W190329 06:26:28.877854 216582 storage/store.go:3577  [s4,r1/2:/M{in-ax}] raft error: node 1 claims to not contain store 1 for replica (n1,s1):1: store 1 was not found
W190329 06:26:28.879782 216580 storage/raft_transport.go:583  while processing outgoing Raft queue to node 1: store 1 was not found:
W190329 06:26:28.949602 215243 gossip/gossip.go:1501  [n3] first range unavailable; trying remaining resolvers
I190329 06:26:28.951469 222769 gossip/client.go:128  [n3] started gossip client to 127.0.0.1:46603
W190329 06:26:28.986701 222818 storage/raft_transport.go:281  unable to accept Raft message from (n5,s5):3: no handler registered for (n1,s1):1
W190329 06:26:28.991724 222829 storage/raft_transport.go:281  unable to accept Raft message from (n5,s5):3: no handler registered for (n2,s2):4
W190329 06:26:28.999740 216550 storage/raft_transport.go:281  unable to accept Raft message from (n5,s5):3: no handler registered for (n4,s4):2
W190329 06:26:29.021727 222855 storage/raft_transport.go:436  no handler found for store 5 in response range_id:1 from_replica:<node_id:1 store_id:1 replica_id:1 > to_replica:<node_id:5 store_id:5 replica_id:3 > union:<error:<message:"store 1 was not found" transaction_restart:NONE origin_node:0 detail:<store_not_found:<store_id:1 > > now:<> > > 
W190329 06:26:29.038021 216465 storage/raft_transport.go:436  no handler found for store 5 in response range_id:1 from_replica:<node_id:4 store_id:4 replica_id:2 > to_replica:<node_id:5 store_id:5 replica_id:3 > union:<error:<message:"store 4 was not found" transaction_restart:NONE origin_node:0 detail:<store_not_found:<store_id:4 > > now:<> > > 
W190329 06:26:29.066854 222875 storage/raft_transport.go:436  no handler found for store 5 in response range_id:1 from_replica:<node_id:2 store_id:2 replica_id:4 > to_replica:<node_id:5 store_id:5 replica_id:3 > union:<error:<message:"store 2 was not found" transaction_restart:NONE origin_node:0 detail:<store_not_found:<store_id:2 > > now:<> > > 
W190329 06:26:29.074294 215832 gossip/gossip.go:1496  [n6] no incoming or outgoing connections
W190329 06:26:29.079236 222853 storage/raft_transport.go:583  while processing outgoing Raft queue to node 2: rpc error: code = Unavailable desc = transport is closing:
--- FAIL: TestStoreRangeRemoveDead (47.94s)
    <autogenerated>:1: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1 (n4,s4):2 (n5,s5):3 (n2,s2):4]
        goroutine 214795 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc00a31e9c0, 0x4aefe40)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4b648a0, 0xc000ba4a00, 0xc00a31e930)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc000ba4a00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2940 +0xaef
        testing.tRunner(0xc000ba4a00, 0x3dfc770)
        	/usr/local/go/src/testing/testing.go:827 +0x163
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/6e0d7081547ee76e52c5f798b78ff739feec2d06

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1208785&tab=buildLog

```
I190329 06:44:09.114176 212455 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190329 06:44:10.116015 212455 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190329 06:44:11.126674 212455 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190329 06:44:12.136510 212455 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190329 06:44:13.159699 212455 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190329 06:44:14.163095 212455 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190329 06:44:15.170585 212455 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190329 06:44:16.186327 212455 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190329 06:44:17.202518 212455 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190329 06:44:18.259286 222444 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190329 06:44:18.261675 222440 util/stop/stopper.go:546  quiescing; tasks left:
2      mtc send
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190329 06:44:18.262114 222441 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190329 06:44:18.262359 222441 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190329 06:44:18.262627 222443 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190329 06:44:18.263118 222442 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190329 06:44:18.263434 222442 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190329 06:44:18.263734 222445 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190329 06:44:18.265234 222445 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190329 06:44:18.266967 222440 util/stop/stopper.go:546  quiescing; tasks left:
1      mtc send
W190329 06:44:18.269766 213054 internal/client/txn.go:509  [hb] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (server shutdown)
I190329 06:44:18.270336 213054 storage/node_liveness.go:775  [hb] retrying liveness update after storage.errRetryLiveness: result is ambiguous (server shutdown)
W190329 06:44:18.272099 213054 internal/client/txn.go:509  [hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W190329 06:44:18.278749 213054 storage/node_liveness.go:458  [hb] failed node liveness heartbeat: node unavailable; try another peer
W190329 06:44:18.559316 213105 gossip/gossip.go:1496  [n5] no incoming or outgoing connections
--- FAIL: TestStoreRangeRemoveDead (46.58s)
    <autogenerated>:1: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1]
        goroutine 212455 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc0009423c0, 0x4a8eea0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4b02840, 0xc001a56e00, 0xc0009420c0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc001a56e00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2940 +0xadb
        testing.tRunner(0xc001a56e00, 0x3dbc258)
        	/usr/local/go/src/testing/testing.go:827 +0x163
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/23f9707873abbd2de91a42055535529d7ff296ce

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1210797&tab=buildLog

```
I190330 07:21:27.254158 217211 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190330 07:21:28.276367 217211 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190330 07:21:29.284437 217211 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190330 07:21:30.311363 217211 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190330 07:21:31.321502 217211 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190330 07:21:32.331632 217211 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190330 07:21:33.350822 217211 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190330 07:21:34.394936 217211 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190330 07:21:35.409322 217211 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190330 07:21:36.424933 217211 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190330 07:21:37.429244 217211 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190330 07:21:38.433190 217211 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190330 07:21:39.436625 217211 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190330 07:21:40.447280 217211 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190330 07:21:41.451711 217211 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190330 07:21:42.453519 217211 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190330 07:21:43.457462 217211 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190330 07:21:44.461278 217211 storage/client_raft_test.go:2940  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190330 07:21:45.470684 228229 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190330 07:21:45.471091 228096 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190330 07:21:45.471489 228097 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190330 07:21:45.473415 228099 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190330 07:21:45.475225 228228 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190330 07:21:45.475707 228098 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190330 07:21:45.477438 228229 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
--- FAIL: TestStoreRangeRemoveDead (46.93s)
    <autogenerated>:1: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1]
        goroutine 217211 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc001a3df50, 0x4a9c280)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4b0ff20, 0xc006c1e200, 0xc001a3df20)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc006c1e200)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2940 +0xadb
        testing.tRunner(0xc006c1e200, 0x3dc5bc8)
        	/usr/local/go/src/testing/testing.go:827 +0x163
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/a039a93a5cc6eb3f395ceb6f7dc8030985dccc29

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1213145&tab=buildLog

```
I190401 06:31:28.273813 208664 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190401 06:31:29.283276 208664 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190401 06:31:30.309296 208664 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190401 06:31:31.333386 208664 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190401 06:31:32.336634 208664 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190401 06:31:33.343022 208664 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190401 06:31:34.345412 208664 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190401 06:31:35.349473 208664 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190401 06:31:36.360400 208664 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190401 06:31:37.367263 208664 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190401 06:31:38.372560 208664 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190401 06:31:39.378596 208664 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190401 06:31:40.405398 208664 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190401 06:31:41.411671 208664 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190401 06:31:42.417539 208664 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190401 06:31:43.446114 208664 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190401 06:31:44.458906 208664 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190401 06:31:45.485786 219289 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190401 06:31:45.489752 219284 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190401 06:31:45.494759 219285 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190401 06:31:45.496731 219285 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
I190401 06:31:45.502736 219286 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190401 06:31:45.505379 219287 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190401 06:31:45.506629 219288 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190401 06:31:45.512842 219289 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
--- FAIL: TestStoreRangeRemoveDead (46.69s)
    <autogenerated>:1: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1]
        goroutine 208664 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc00d079c80, 0x4aa84c0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4b1c180, 0xc00181b600, 0xc00d079c50)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc00181b600)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2942 +0xadb
        testing.tRunner(0xc00181b600, 0x3dcfcf8)
        	/usr/local/go/src/testing/testing.go:827 +0x163
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/877ebd1ece299b9ee621aa0d091657621593d844

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1216457&tab=buildLog

```
I190402 06:58:08.531783 208319 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190402 06:58:09.557724 208319 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190402 06:58:10.563021 208319 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190402 06:58:11.597294 208319 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190402 06:58:12.599650 208319 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190402 06:58:13.616989 208319 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190402 06:58:14.654202 208319 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190402 06:58:15.665689 208319 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190402 06:58:16.677177 208319 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190402 06:58:17.680752 208319 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190402 06:58:18.722613 208319 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190402 06:58:19.726080 208319 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190402 06:58:20.743717 208319 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190402 06:58:21.778056 208319 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190402 06:58:22.783633 208319 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190402 06:58:23.796371 218565 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190402 06:58:23.796678 218565 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190402 06:58:23.796996 218480 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190402 06:58:23.797379 218481 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190402 06:58:23.797653 218481 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
I190402 06:58:23.797910 218482 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190402 06:58:23.798273 218484 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190402 06:58:23.798523 218483 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190402 06:58:23.811278 218484 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
W190402 06:58:24.099239 208906 gossip/gossip.go:1496  [n4] no incoming or outgoing connections
--- FAIL: TestStoreRangeRemoveDead (46.53s)
    <autogenerated>:1: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1]
        goroutine 208319 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc00bbbc060, 0x4aaeca0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4b22aa0, 0xc00ff64200, 0xc00bbbc030)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc00ff64200)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2942 +0xadb
        testing.tRunner(0xc00ff64200, 0x3dd4e50)
        	/usr/local/go/src/testing/testing.go:827 +0x163
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/1cbf3680129e47bd310640bf32b665662f30faa9

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1218740&tab=buildLog

```
I190403 08:19:46.302720 208022 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190403 08:19:47.331058 208022 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190403 08:19:48.347560 208022 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190403 08:19:49.352628 208022 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190403 08:19:50.358584 208022 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190403 08:19:51.372957 208022 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190403 08:19:52.378820 208022 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190403 08:19:53.381581 208022 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190403 08:19:54.386992 208022 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190403 08:19:55.390592 218206 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190403 08:19:55.391093 218204 util/stop/stopper.go:546  quiescing; tasks left:
2      mtc send
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190403 08:19:55.391458 218204 util/stop/stopper.go:546  quiescing; tasks left:
2      mtc send
1      [async] intent_resolver_ir_batcher
I190403 08:19:55.391936 218205 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190403 08:19:55.392806 218207 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190403 08:19:55.393186 218208 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190403 08:19:55.394134 218208 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_gc_batcher
I190403 08:19:55.394735 218209 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190403 08:19:55.400059 218205 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190403 08:19:55.406241 218204 util/stop/stopper.go:546  quiescing; tasks left:
1      mtc send
W190403 08:19:55.416794 208155 internal/client/txn.go:509  [hb] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (server shutdown)
I190403 08:19:55.417087 208155 storage/node_liveness.go:775  [hb] retrying liveness update after storage.errRetryLiveness: result is ambiguous (server shutdown)
W190403 08:19:55.422370 208155 internal/client/txn.go:509  [hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W190403 08:19:55.424873 208155 storage/node_liveness.go:458  [hb] failed node liveness heartbeat: node unavailable; try another peer
--- FAIL: TestStoreRangeRemoveDead (46.79s)
    <autogenerated>:1: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1]
        goroutine 208022 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc001105e00, 0x4aac100)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4b1fe40, 0xc0036aad00, 0xc001105da0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc0036aad00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2942 +0xadb
        testing.tRunner(0xc0036aad00, 0x3dd25f8)
        	/usr/local/go/src/testing/testing.go:827 +0x163
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x65a

```

SHA: https://github.com/cockroachdb/cockroach/commits/759f4076aced721bdfbb16bafb55f4662182e519

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestStoreRangeRemoveDead PKG=github.com/cockroachdb/cockroach/pkg/storage TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1221219&tab=buildLog

```
I190404 07:24:03.627956 206165 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190404 07:24:04.677953 206165 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190404 07:24:05.681940 206165 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190404 07:24:06.706423 206165 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190404 07:24:07.724998 206165 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190404 07:24:08.732865 206165 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190404 07:24:09.735216 206165 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190404 07:24:10.746333 206165 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190404 07:24:11.756457 206165 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190404 07:24:12.759771 206165 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190404 07:24:13.763076 206165 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190404 07:24:14.787181 206165 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190404 07:24:15.793854 206165 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190404 07:24:16.802583 206165 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190404 07:24:17.814840 206165 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190404 07:24:18.819590 206165 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190404 07:24:19.836724 206165 storage/client_raft_test.go:2942  SucceedsSoon: expected 5 replicas; have [(n1,s1):1]
I190404 07:24:20.850805 216097 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190404 07:24:20.853774 216092 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190404 07:24:20.854272 216093 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190404 07:24:20.854629 216094 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190404 07:24:20.855489 216094 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
I190404 07:24:20.855778 216095 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
I190404 07:24:20.856327 216096 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
W190404 07:24:21.020241 206935 gossip/gossip.go:1496  [n5] no incoming or outgoing connections
W190404 07:24:21.023943 206353 gossip/gossip.go:1496  [n2] no incoming or outgoing connections
--- FAIL: TestStoreRangeRemoveDead (46.47s)
    <autogenerated>:1: condition failed to evaluate within 45s: expected 5 replicas; have [(n1,s1):1]
        goroutine 206165 [running]:
        runtime/debug.Stack(0xa7a358200, 0xc001110780, 0x4ac34c0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0xb5
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x4b372c0, 0xc001bd0500, 0xc001110750)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:49 +0x16c
        github.com/cockroachdb/cockroach/pkg/storage_test.TestStoreRangeRemoveDead(0xc001bd0500)
        	/go/src/github.com/cockroachdb/cockroach/pkg/storage/client_raft_test.go:2942 +0xaef
        testing.tRunner(0xc001bd0500, 0x3ddf140)
        	/usr/local/go/src/testing/testing.go:827 +0x163
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:878 +0x65a

```

--- a/pkg/storage/client_raft_test.go
+++ b/pkg/storage/client_raft_test.go
@@ -32,7 +32,6 @@ import (
 	"go.etcd.io/etcd/raft/raftpb"
 
 	"github.com/cockroachdb/cockroach/pkg/base"
-	"github.com/cockroachdb/cockroach/pkg/config"
 	"github.com/cockroachdb/cockroach/pkg/gossip"
 	"github.com/cockroachdb/cockroach/pkg/internal/client"
 	"github.com/cockroachdb/cockroach/pkg/keys"
@@ -2856,87 +2855,6 @@ func TestStoreRangeMoveDecommissioning(t *testing.T) {
 			if r.NodeID == decommingNodeID {
 				return errors.Errorf("expected replica to be moved off node %d, but got replicas %v",
 					decommingNodeID, curReplicas)
-			}
-		}
-		return nil
-	})
-}
-
-// TestStoreRangeRemoveDead verifies that if a store becomes dead, the
-// ReplicateQueue will notice and remove any replicas on it.
-func TestStoreRangeRemoveDead(t *testing.T) {
-	defer leaktest.AfterTest(t)()
-	sc := storage.TestStoreConfig(nil)
-	sc.TestingKnobs.DisableReplicaRebalancing = true
-	mtc := &multiTestContext{storeConfig: &sc}
-	mtc.timeUntilStoreDead = storage.TestTimeUntilStoreDead
-	defer mtc.Stop()
-
-	zone := config.DefaultSystemZoneConfig()
-	mtc.Start(t, int(zone.NumReplicas+1))
-
-	var nonDeadStores []*storage.Store
-	for i, s := range mtc.stores {
-		if i == 1 {
-			// Skip the dead store.
-			continue
-		}
-		nonDeadStores = append(nonDeadStores, s)
-	}
-
-	// Create a goroutine to gossip store capacity info periodically.
-	go func() {
-		tickerDur := storage.TestTimeUntilStoreDead / 2
-		ticker := time.NewTicker(tickerDur)
-		defer ticker.Stop()
-
-		for {
-			select {
-			case <-ticker.C:
-				mtc.manualClock.Increment(int64(tickerDur))
-
-				// Keep gossiping the stores, excepting the dead store.
-				for _, s := range nonDeadStores {
-					if err := s.GossipStore(context.Background(), false /* useCached */); err != nil {
-						panic(err)
-					}
-				}
-				// Force the repair queues on all alive stores to run.
-				for _, s := range nonDeadStores {
-					s.ForceReplicationScanAndProcess()
-				}
-
-			case <-mtc.stoppers[0].ShouldStop():
-				return
-			}
-		}
-	}()
-
-	// Wait for up-replication.
-	testutils.SucceedsSoon(t, func() error {
-		replicas := getRangeMetadata(roachpb.RKeyMin, mtc, t).Replicas
-		if len(replicas) == int(zone.NumReplicas) {
-			return nil
-		}
-		return errors.Errorf("expected %d replicas; have %+v", zone.NumReplicas, replicas)
-	})
-
-	// Stop a store which will be rebalanced away from. We can't use the very first
-	// one since getRangeMetadata is hard-coded to query that one, so we'll use the
-	// one after that.
-	deadStoreID := mtc.stores[1].StoreID() // 2
-	mtc.stopStore(1)
-	// The mtc asks us to restart this store before stopping the mtc.
-	defer mtc.restartStoreWithoutHeartbeat(1)
-
-	testutils.SucceedsSoon(t, func() error {
-		replicas := getRangeMetadata(roachpb.RKeyMin, mtc, t).Replicas
-		if len(replicas) != int(zone.NumReplicas) {
-			return errors.Errorf("expected %d replicas; have %+v", zone.NumReplicas, replicas)
-		}
-		for _, r := range replicas {
-			if r.StoreID == deadStoreID {
-				return errors.Errorf("expected store %d to be replaced; have %+v", r.StoreID, replicas)
 			}
 		}
 		return nil

